Tests/StreamChatTests/Repositories/PollsRepository_Tests.swift

```diff
@@ -0,0 +1,479 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChat
+@testable import StreamChatTestTools
+import XCTest
+
+final class PollsRepository_Tests: XCTestCase {
+    var apiClient: APIClient_Spy!
+    var database: DatabaseContainer!
+    var repository: PollsRepository!
+
+    override func setUp() {
+        super.setUp()
+
+        apiClient = APIClient_Spy()
+        database = DatabaseContainer_Spy()
+        repository = PollsRepository(database: database, apiClient: apiClient)
+    }
+
+    override func tearDown() {
+        apiClient.cleanUp()
+
+        apiClient = nil
+        repository = nil
+        database = nil
+
+        super.tearDown()
+    }
+
+    // MARK: - Poll creation
+    
+    func test_createPoll_whenSuccessfull() {
+        let completionCalled = expectation(description: "completion called")
+        let pollName = "Test Poll"
+        
+        repository.createPoll(name: pollName) { result in
+            XCTAssertNil(result.error)
+            completionCalled.fulfill()
+        }
+        
+        let payload = XCTestCase().dummyPollPayload()
+        let response = PollPayloadResponse(duration: "", poll: payload)
+        apiClient.test_simulateResponse(.success(response))
+        
+        wait(for: [completionCalled], timeout: defaultTimeout)
+        let referenceEndpoint: Endpoint<PollPayloadResponse> = .createPoll(
+            createPollRequest: .init(name: pollName)
+        )
+        XCTAssertEqual(apiClient.request_endpoint, AnyEndpoint(referenceEndpoint))
+        XCTAssertEqual(payload.name, pollName)
+    }
+    
+    func test_createPoll_whenFailure() {
+        let completionCalled = expectation(description: "completion called")
+        let pollName = "Test Poll"
+        
+        repository.createPoll(name: pollName) { result in
+            XCTAssertNotNil(result.error)
+            completionCalled.fulfill()
+        }
+        
+        let error = TestError()
+        apiClient.test_simulateResponse(Result<PollPayloadResponse, Error>.failure(error))
+        
+        wait(for: [completionCalled], timeout: defaultTimeout)
+        let referenceEndpoint: Endpoint<PollPayloadResponse> = .createPoll(
+            createPollRequest: .init(name: pollName)
+        )
+        XCTAssertEqual(apiClient.request_endpoint, AnyEndpoint(referenceEndpoint))
+    }
+    
+    // MARK: - Closing poll
+    
+    func test_closePoll_whenSuccessfull() {
+        let completionCalled = expectation(description: "completion called")
```

Comment:
Using expectations is much better and familiar to everyone 👍 It would be nice that we kept consistency and not use AssertAsync in someplaces and others expecations (unless it is necessary to use AssertAsync because of bad design)